src/receive_imf.rs

```diff
@@ -195,9 +195,14 @@ pub(crate) async fn receive_imf_inner(
 
     // For the case if we missed a successful SMTP response.
     if !incoming {
+        let self_addr = context.get_primary_self_addr().await?;
         context
             .sql
-            .execute("DELETE FROM smtp WHERE rfc724_mid=?", (rfc724_mid_orig,))
+            .execute(
+                "DELETE FROM smtp \
+                WHERE rfc724_mid=?1 AND (recipients LIKE ?2 OR recipients LIKE ('% ' || ?2))",
```

Comment:
This still can break sending already existing messages in the `smtp` table (because they are not split), but this is a temporary problem and may occur only in case of SMTP timeouts

Reply:
Added a migration instead